src/librustc_mir/borrow_check.rs

```diff
@@ -446,54 +446,78 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> MirBorrowckCtxt<'c, 'b, 'a, 'gcx, 'tcx>
 
         let (sd, rw) = kind;
 
-        // Check permissions
-        self.check_access_permissions(lvalue_span, rw);
-
-        self.each_borrow_involving_path(
-            context, (sd, lvalue_span.0), flow_state, |this, _index, borrow, common_prefix| {
-                match (rw, borrow.kind) {
-                    (Read(_), BorrowKind::Shared) => {
-                        Control::Continue
-                    }
-                    (Read(kind), BorrowKind::Unique) |
-                    (Read(kind), BorrowKind::Mut) => {
-                        match kind {
-                            ReadKind::Copy =>
-                                this.report_use_while_mutably_borrowed(
-                                    context, lvalue_span, borrow),
-                            ReadKind::Borrow(bk) => {
-                                let end_issued_loan_span =
-                                    flow_state.borrows.base_results.operator().opt_region_end_span(
-                                        &borrow.region);
-                                this.report_conflicting_borrow(
-                                    context, common_prefix, lvalue_span, bk,
-                                    &borrow, end_issued_loan_span)
-                            }
+        fn check_borrows_readability_or_writability<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a>(
```

Comment:
If we are suppressing the borrows, we shouldn't need any changes to this function.